common/src/main/java/org/apache/drill/common/util/ProtobufPatcher.java

```diff
@@ -57,15 +60,15 @@ public static synchronized void patch() {
   private static void patchByteString() {
     try {
       ClassPool classPool = getClassPool();
-      CtClass byteString = classPool.get("com.google.protobuf.ByteString");
+      CtClass byteString = classPool.get(protobufPackage + "ByteString");
```

Comment:
I'm afraid it will break JDBC client. As you know, we shade protobuf jars into JDBC driver and relocate them from `com.google` to `oadd.com.google`.
Shade plugin traverses all classes and replaces packages with updated ones. String literal with full class name also will be updated, but with your change, it breaks this behavior, so classes wouldn't be shades.
See the description of this behavior here: https://issues.apache.org/jira/browse/MSHADE-156. 

After fixing this issue, please ensure that the custom authenticator works correctly with your changes (this one should break it).

Reply:
Verified with different JDBC clients w/ and w/o security, with pam4j and custom auth module, everything is working fine. Also, by decompilation of ProtobufPatcher class from the driver I can see that all strings with package names were renamed with addition of "oadd." prefix.

Reply:
Thanks for checking. Looks like I observed this issue for another case of using a fully qualified name.